docs/ECMAScriptModules.md

```diff
@@ -38,4 +38,51 @@ import.meta.jest.useFakeTimers();
 // jest === import.meta.jest => true
 ```
 
+Additionally, since ESM evaluates static `import` statements before looking at the code, hoisting on `jest.mock` calls that happens in CJS modules won't work in ESM. To `mock` modules in ESM, you need to use dynamic `import()` after `jest.mock` calls to load the mocked modules, same applies to modules which have to load the mocked modules.
```

Comment:
why do we talk about "dynamic `import()`, and then go on to use `require`?

Reply:
I've made the change to also mention `require` there, with the example now showing both `require` and dynamic `import()` usage. Personally I prefer `require` for CJS modules but I think it's better to show both on the docs so the users aren't going to think that either is more preferable.